Runner/JobManager.py

```diff
@@ -0,0 +1,334 @@
+
+        full_sas_url_input = 'https://{}.blob.core.windows.net/{}?{}'.format(
+            blob_client.account_name,
+            input_container_name,
+            Utils.get_container_sas_token(
+                blob_client,
+                input_container_name,
+                ContainerPermissions.READ +
+                ContainerPermissions.LIST))
+        full_sas_url_output = 'https://{}.blob.core.windows.net/{}?{}'.format(
+            blob_client.account_name,
+            output_container_name,
+            Utils.get_container_sas_token(
+                blob_client,
+                output_container_name,
+                ContainerPermissions.READ +
+                ContainerPermissions.LIST +
+                ContainerPermissions.WRITE))
+
+        # Set the storage info for the container.
+        self.storage_info = Utils.StorageInfo(
+            input_container_name,
+            output_container_name,
+            full_sas_url_input,
+            full_sas_url_output)
+
+        # Upload the asset file that will be rendered and
+        scenefile = ctm.get_scene_file(self.parameters_file)
+        for file in os.listdir("Assets"):
+            if scenefile == file:
+                filePath = Path("Assets/" + file)
+                await loop.run_in_executor(None, functools.partial(Utils.upload_file_to_container, blob_client, input_container_name, filePath))
+
+    async def check_expected_output(self, batch_service_client):
+        """
+        Checks to see if the the job expected output is correct
+
+        """
+        loop = asyncio.get_event_loop()
+
+        if self.job_status.job_state == Utils.JobState.COMPLETE:
+            self.job_status = await loop.run_in_executor(None, Utils.check_task_output, batch_service_client, self.job_id, self.expected_output)
+
+    def check_pool_resize_error(self, pool):
+        if pool.allocation_state.value == "steady" and pool.resize_errors is not None:
+            self.job_status = Utils.JobStatus(Utils.JobState.POOL_FAILED,
+                                              "Job failed to start since the pool [{}] failed to allocate any TVMs due to error [Code: {}, message {}]."
+                                              .format(self.pool_id, pool.resize_errors[0].code, pool.resize_errors[0].message))
+            print("POOL {} FAILED TO ALLOCATE".format(self.pool_id))
+            return True
+        return False
+
+    def check_timeout(self, timeout):
+        timeout_expiration = datetime.datetime.now() + timeout
+        return datetime.datetime.now() <= timeout_expiration
+
+    def check_pool_state(self, batch_service_client, timeout):
+        pool = batch_service_client.pool.get(self.pool_id)
+
+        # wait for pool to come up
+        while pool.allocation_state.value == "resizing" and self.check_timeout(
+                timeout):
+            time.sleep(10)
```

Comment:
:bulb: What is the purpose of `sleep` here? (Please add comment)

Reply:
Queries the pool status every 10 seconds instead of every second. I don't want to query the service every second because it would be too noisy.